fim_pre_processing.sh

```diff
@@ -0,0 +1,244 @@
+#!/bin/bash -e
+:
+usage ()
+{
+    echo
+    echo 'Produce FIM hydrofabric datasets for unit and branch scale.'
+    echo 'NOTE: this file runs hucs and branches AND post processing.'    
```

Comment:
"AND post processing" - isn't that taken care of by `fim_post_processing.sh`?

Reply:
hummm... ya, I need to say that better. I am trying to say that it takes care of fim_post_processing, processing all of the huc(s), then runs fim_post_processing.